search/searcher/search_disjunction.go

```diff
@@ -46,16 +46,33 @@ func optionsDisjunctionOptimizable(options search.SearcherOptions) bool {
 func newDisjunctionSearcher(ctx context.Context, indexReader index.IndexReader,
 	qsearchers []search.Searcher, min float64, options search.SearcherOptions,
 	limit bool) (search.Searcher, error) {
+
+	// The KNN Searcher optimization is a necessary pre-req for KNN Searchers, not an
+	// optional optimization like for, say term searchers.
+	// It's an optimization in the sense that it's more optimal to batch calls
+	// Hence, this optimization should be invoked for KNN searchers irrespective
+	// of whether there are other searchers in the disjunction or not.
+
+	optimizedKNNSearchers, err := optimizeKNNSearcher(ctx, "", indexReader, qsearchers, options)
+	if err != nil {
+		return nil, err
+	}
+
 	// attempt the "unadorned" disjunction optimization only when we
 	// do not need extra information like freq-norm's or term vectors
 	// and the requested min is simple
 	if len(qsearchers) > 1 && min <= 1 &&
 		optionsDisjunctionOptimizable(options) {
 		rv, err := optimizeCompositeSearcher(ctx, "disjunction:unadorned",
 			indexReader, qsearchers, options)
-		if err != nil || rv != nil {
+		if err != nil {
 			return rv, err
 		}
+		if rv != nil && optimizedKNNSearchers != nil && len(optimizedKNNSearchers) != 0 {
+			qsearchers = make([]search.Searcher, 1+len(optimizedKNNSearchers))
```

Comment:
Why `1+`? Also if this is necessary suffix `+1` instead.

Reply:
I'm assuming the latter point is just for coding style :)